include/xtensor/xmath.hpp

```diff
@@ -2958,6 +2958,31 @@ namespace detail {
         return interp(x, xp, fp, fp[0], fp[fp.size() - 1]);
     }
 
+    /**
+     * @brief Returns the covariance matrix
+     * 
+     * @param x Source matrix
+     */
+    template <class E1>
+    inline auto cov(const E1 &x)
+    {
+        using value_type = typename E1::value_type;
+
+        XTENSOR_ASSERT( x.dimension() == 2 );
+
+        auto s = x.shape();
+        auto covar = eval(zeros<value_type>({s[1], s[1]}));
+
+        for (auto i = 0; i < s[1]; i++)
+        {
+            auto xi = strided_view(x, { all(), range(i, i + 1) });
```

Comment:
using `view` should be faster:
```cpp
auto xi = xt::view(x, all(), range(i, i+1));
```

Reply:
view has not been used in xmath so far, therefore i thought i go with strided view as used in this file

Reply:
including xview in xmath actually raises a lot of error